### PR TITLE
PLT-5436: Render channel links in the channel header

### PR DESCRIPTION
**Pull request by panditsavitags**
#### Summary
Channel Header is now linked to the specific channel if the text following the (~) matches any of the display_names.

#### Ticket Link
[issue-7257](https://github.com/mattermost/mattermost-server/issues/7257)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [ ] Has server changes (please link)
- [ ] Has redux changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

**Comment by panditsavitags** [comment]
@jasonblais , Please review the PR

**Comment by sudheerDev** [comment]
@panditsavitags One thing here. Just add only your commits. From the looks of it last three commits are only needed. So you can discard the rest of commits from PR.

**Comment by panditsavitags** [comment]
Sure @esethna , thanks :) Please merge this PR.

File: components/channel_header/channel_header.jsx [hunk 1]
```diff
@@ -716,7 +718,7 @@ export default class ChannelHeader extends React.Component {
                         {dmHeaderTextStatus}
                         <span
                             onClick={Utils.handleFormattedTextClick}
-                            dangerouslySetInnerHTML={{__html: TextFormatting.formatText(channel.header, {singleline: true, mentionHighlight: false, siteURL: getSiteURL()})}}
+                            dangerouslySetInnerHTML={{__html: TextFormatting.formatText(channel.header, {singleline: true, mentionHighlight: false, siteURL: getSiteURL(), channelNamesMap: ChannelStore.getChannelNamesMap(), team: TeamStore.getCurrent(), atMentions: true})}}
```
Comment (saturninoabril) [inline components/channel_header/channel_header.jsx]: This line is very long and not easy to read. Could you please shorten it and make more readable? Thanks!

File: utils/text_formatting.jsx [hunk 2]
```diff
@@ -187,7 +187,7 @@ function autolinkChannelMentions(text, tokens, channelNamesMap, team) {
         }
 
         tokens.set(alias, {
-            value: `<a class="mention-link" href="${href}" data-channel-mention="${channelName}">~${displayName}</a>`,
+            value: `<a class="mention-link" href="${href}" data-channel-mention="${channelName}">${displayName}</a>`,
```
[the remark this answers is not in the document]
Comment (panditsavitags) [reply utils/text_formatting.jsx]: @saturninoabril , Sure i will make the changes and push the code, thank you for your advice.

**Comment by panditsavitags** [comment]
@esethna , This was not mentioned in the requirement, sure i will get this done. Thank you :)

**Comment by panditsavitags** [comment]
@saturninoabril , I have made the required changes and pushed it, please review and merge the PR

**Comment by panditsavitags** [comment]
@saturninoabril , please check the update and help in merging process

**Comment by saturninoabril** [comment]
@panditsavitags Please help rebase to latest then we're good to merge this. Thanks!

**Comment by panditsavitags** [comment]
@saturninoabril , please merge it

**Pull request merged**